tests/unit/test_artifactory_path.py

```diff
@@ -736,6 +738,92 @@ def _mock_properties_response(self):
         )
         return path
 
+    @responses.activate
+    def test_unlink(self):
+        """
+        Test that folder/file unlink works
+        """
+        path = ArtifactoryPath(
+            "http://artifactory.local/artifactory/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        constructed_url = (
+            "http://artifactory.local/artifactory"
+            "/api/storage"
+            "/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        responses.add(
+            responses.GET,
+            constructed_url,
+            status=200,
+            json=self.file_stat,
+        )
+
+        responses.add(
+            responses.DELETE,
+            str(path),
+            status=200,
+        )
+
+        path.unlink()
+
+    @responses.activate
+    def test_unlink_raises_not_found(self):
+        """
+        Test that folder/file unlink raises OSError if file does not exist
+        """
+        path = ArtifactoryPath(
+            "http://artifactory.local/artifactory/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        constructed_url = (
+            "http://artifactory.local/artifactory"
+            "/api/storage"
+            "/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        responses.add(
+            responses.GET,
+            constructed_url,
+            status=404,
+            body="Unable to find item",
+        )
+        with self.assertRaises(OSError) as context:
+            path.unlink()
+
+        self.assertTrue("No such file or directory" in context.exception.strerror)
+
+    @responses.activate
+    def test_unlink_raises_on_404(self):
+        """
+        Test that folder/file unlink raises exception if we checked that file
+        exsists and we still get 404. This is a result of permission issue
+        """
+        path = ArtifactoryPath(
+            "http://artifactory.local/artifactory/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        constructed_url = (
+            "http://artifactory.local/artifactory"
+            "/api/storage"
+            "/ext-release-local/org/company/tool/1.0/tool-1.0.tar.gz"
+        )
+        responses.add(
+            responses.GET,
+            constructed_url,
+            status=200,
+            json=self.file_stat,
+        )
+
+        responses.add(
+            responses.DELETE,
+            str(path),
+            status=404,
+        )
+
+        with self.assertRaises(ArtifactoryException) as context:
+            path.unlink()
+
+        self.assertTrue(
+            "insufficient Artifactory privileges" in context.exception.strerror
```

Comment:
Why do we need `strerror  at all, we can call `str(exception)` directly.

Reply:
strerror in `OSError` relates to [os.strerror](https://docs.python.org/3/library/os.html#os.strerror), we should not add the property to ArtifactoryException, because it doesn't relate to it